app/controllers/spree/api/images_controller_decorator.rb

```diff
@@ -0,0 +1,15 @@
+Spree::Api::ImagesController.class_eval do
+  def update_product_image
+    @product = Spree::Product.find(params[:product_id])
+    authorize! :update, @product
+
+    if @product.images.first.nil?
```

Comment:
What about `@product.images.empty?`?

Reply:
Just tried this out, and it doesn't work exactly as expected with `@product.images.empty?`. I can't remember why now, but it was something to do with the way products and variants can all have multiple images and they get returned in a strange way.

Reply:
😞 👍